libraries/consul_config.rb

```diff
@@ -82,7 +85,10 @@ class ConsulConfig < Chef::Resource
       attribute(:retry_interval_wan, kind_of: String)
       attribute(:retry_join, kind_of: Array)
       attribute(:retry_join_wan, kind_of: Array)
+      attribute(:retry_join_ec2, kind_of: [Hash, Mash])
```

Comment:
Following the alphabet order, then it should be defined _before_ `retry_join_wan` ;)